fix(span): `f64::content_eq` return `false` for `0` and `-0`

overlookmotel opened the pull request:
Fixes #8982.

Previously `0f64.content_eq(-0f64)` returned `true`. Now it returns `false`.

This also affects the behavior for `NaN`. Previously `f64::NAN.content_eq(f64::NAN)` returned `false`. Now it returns `true`. But it's complicated:

```rs
f64::NAN.content_eq(f64::NAN) == true
f64::NAN.content_eq(-f64::NAN) == false
f64::NAN.content_eq(--f64::NAN) == true
```

This does *not* align with JS's `Object.is` which returns `true` for *any* two `NaN` values.

If this matters, we could instead implement `f64::content_eq` as:

```rs
if self.is_nan() && other.is_nan() {
    true
} else {
    self.to_bits() == other.to_bits()
}
```

But this generates quite a lot of assembly for all `f64` values, just to cover this small edge case with `NaN`: https://godbolt.org/z/q9zYodn4e

So I suggest that we go with it as in this PR for now, and see if `NaN` causes us problems in reality or not.

overlookmotel commented:
* **#9007** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9007?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9007?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=9007).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **Feb 10, 7:46 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Feb 10, 7:46 AM EST**: A user added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

overlookmotel committed:
fix(span): `f64::content_eq` return `false` for `0` and `-0` (#9007)

Fixes #8982.

Previously `0f64.content_eq(-0f64)` returned `true`. Now it returns `false`.

This also affects the behavior for `NaN`. Previously `f64::NAN.content_eq(f64::NAN)` returned `false`. Now it returns `true`. But it's complicated:

```rs
f64::NAN.content_eq(f64::NAN) == true
f64::NAN.content_eq(-f64::NAN) == false
f64::NAN.content_eq(--f64::NAN) == true
```

This does *not* align with JS's `Object.is` which returns `true` for *any* two `NaN` values.

If this matters, we could instead implement `f64::content_eq` as:

```rs
if self.is_nan() && other.is_nan() {
    true
} else {
    self.to_bits() == other.to_bits()
}
```

But this generates quite a lot of assembly for all `f64` values, just to cover this small edge case with `NaN`: https://godbolt.org/z/q9zYodn4e

So I suggest that we go with it as in this PR for now, and see if `NaN` causes us problems in reality or not.
